bezier-rs/lib/src/lib.rs

```diff
@@ -780,14 +851,15 @@ impl Bezier {
 		endpoint_normal_angle < SCALABLE_CURVE_MAX_ENDPOINT_NORMAL_ANGLE
 	}
 
-	/// Split the curve into a number of scalable subcurves. This function may introduce gaps if subsections of the curve are not reducible.
+	/// Returns a tuple of the scalable subcurves and the corresponding `t` values that were used to split the curve.
+	/// This function may introduce gaps if subsections of the curve are not reducible.
 	/// The function takes the following parameter:
 	/// - `step_size` - Dictates the granularity at which the function searches for reducible subcurves. The default value is `0.01`.
 	///   A small granularity may increase the chance the function does not introduce gaps, but will increase computation time.
-	pub fn reduce(&self, step_size: Option<f64>) -> Vec<Bezier> {
+	fn reduced_curves_and_t_values(&self, step_size: Option<f64>) -> (Vec<Bezier>, Vec<f64>) {
```

Comment:
Would it make sense to return a Vec of tuples rather than a tuple of Vecs? `-> Vec<(Bezier, f64)>`

Reply:
That wouldn't work because there's one more `f64` than `Beziers`, since each `Bezier` is the curve between every pair of `f64`s. I could do `-> Vec<Bezier, [f64; s]>`, but I'm not sure if the duplicate data is worth it or not.

Reply:
Ok, I wasn't sure if they were parallel Vectors or not. Fine as is.